Feature/artesca 15250 title props

JeanMarcMilletScality opened the pull request:
Add loading
Add title props
Add colorSet props

JeanMarcMilletScality committed:
Handle last props for bar chart:
Add header props (title, secondaryTitle, rightTitle and helpTooltip)
Add loading props

Improve style
Add tests for loading and header

JeanMarcMilletScality committed:
Add export for Barchart
Create hook with all compute fn use
Add loading
Add const for hardcode values for bar chart style

bert-e commented:
# Hello jeanmarcmilletscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/core-ui/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @monpote

* @jbwatenbergscality

* @jeanmarcmilletscality

* @hervedombya

JeanMarcMilletScality committed:
feat: add colorSet prop

- Add optional colorSet prop to Barchart component ('default' | 'status')
- Auto-assign status colors based on bar label patterns:
  - Success/Healthy -> theme.statusHealthy
  - Fail/Critical -> theme.statusCritical
  - Warning -> theme.statusWarning
- Custom bar.color always takes priority over colorSet
- Add StatusColors story demonstrating the feature
- Make color property optional in BarchartBars type

JeanMarcMilletScality committed:
Add test for colorSet and update tests

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @hervedombya

* @chengyanjin

* @jeanmarcmilletscality

* @jbwatenbergscality

JeanMarcMilletScality commented:
I will do another PR  separating color adding in its own fn and refactoring the main formatting fn as it is becoming too big and not easily readable

JeanMarcMilletScality committed:
add chartColor in theme + type
Change colorset logic for barchart

ChengYanJin commented on src/lib/components/barchartv2/Barchart.component.tsx:
```diff
@@ -204,11 +205,10 @@ export const Barchart = <T extends BarchartBars>(props: BarchartProps<T>) => {
 
   const { rechartsBars, unitLabel, roundReferenceValue, rechartsData } =
-    useChartData(bars, type, stacked, defaultSort, unitRange, colorSet);
+    useChartData(bars, type, colorSet, stacked, defaultSort, unitRange);
```
oops,

ChengYanJin replied:
We need add `colorSet` to all the stories to avoid the type error.

ChengYanJin commented on src/lib/style/theme.ts:
```diff
@@ -162,6 +171,20 @@ export const lineColor5 = '#E3FF73';
+export type ChartColors = keyof typeof chartColors;
+
+export const chartColors = {
+  lineColor1: '#A14FBF',
```
Nit: we can directly export const chartColors = { lineColor1, lineColor2, ....} since we have already define it before.

JeanMarcMilletScality committed:
use colorset prop for stories and test

ChengYanJin reviewed (approved):
LGTM!!!

JeanMarcMilletScality commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue ARTESCA-15250.

Goodbye jeanmarcmilletscality.




*The following options are set:* **approve**
